app/services/billable_metrics/advanced_aggregations/prorated_unique_count_service.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+module BillableMetrics
+  module AdvancedAggregations
+    class ProratedUniqueCountService < BillableMetrics::Aggregations::UniqueCountService
+      def aggregate(from_datetime:, to_datetime:, options: {})
+        @from_datetime = from_datetime
+        @to_datetime = to_datetime
+
+        # For charges that are pay in advance on billing date we always bill full amount
+        return super if event.nil? && options[:is_pay_in_advance] && !options[:is_current_usage]
+
+        aggregation = compute_prorated_aggregation.ceil(5)
+
+        if options[:is_current_usage]
+          result_without_proration = super
+          handle_current_usage(result_without_proration, aggregation, options[:is_pay_in_advance])
+        else
+          result.aggregation = aggregation
+        end
+
+        result.pay_in_advance_aggregation = compute_pay_in_advance_aggregation
+        result.options = options
+        result.count = result.aggregation
+        result
+      end
+
+      def compute_pay_in_advance_aggregation
+        return BigDecimal(0) unless event
+        return BigDecimal(0) if event.properties.blank?
+
+        result_without_proration = super
+
+        number_of_days = to_datetime.in_time_zone(customer.applicable_timezone) -
+          event.timestamp.in_time_zone(customer.applicable_timezone)
```

Comment:
Layout/MultilineOperationIndentation: Align the operands of an expression in an assignment spanning multiple lines.